common/network-types/src/frame.rs

```diff
@@ -0,0 +1,836 @@
+use crate::errors::NetworkTypeError;
+use crate::errors::NetworkTypeError::{IncompleteFrame, InvalidFrameId, InvalidSegment, ReassemblerClosed};
+use crossbeam_skiplist::SkipSet;
+use dashmap::mapref::entry::Entry;
+use dashmap::DashMap;
+use futures::{Sink, Stream};
+use hopr_platform::time::native::current_time;
+use hopr_primitive_types::prelude::AsUnixTimestamp;
+use serde::{Deserialize, Serialize};
+use std::mem;
+use std::ops::Sub;
+use std::pin::Pin;
+use std::sync::atomic::{AtomicU16, AtomicU64, Ordering};
+use std::sync::OnceLock;
+use std::task::{Context, Poll};
+use std::time::Duration;
+
+fn segment(data: &[u8], mtu: u16, frame_id: u16) -> Vec<Segment> {
+    let chunks = data.chunks(mtu as usize);
+    assert!(chunks.len() < u16::MAX as usize, "data too long");
```

Comment:
There will never be more chunks, since the type of mtu is u16. The assert! is there if the type changes in the future, at which point it should fail hard to force a fix.